clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -58,6 +58,28 @@ static RValue emitBuiltinBitOp(CIRGenFunction &cgf, const CallExpr *e,
   return RValue::get(result);
 }
 
+static mlir::Value makeAtomicFenceValue(CIRGenFunction &cgf,
+                                        const CallExpr *expr,
+                                        cir::SyncScopeKind syncScope) {
+  auto &builder = cgf.getBuilder();
+  mlir::Value orderingVal = cgf.emitScalarExpr(expr->getArg(0));
+
+  auto constOrdering = orderingVal.getDefiningOp<cir::ConstantOp>();
+  if (!constOrdering)
+    llvm_unreachable("NYI: variable ordering not supported");
```

Comment:
Well this is not the intended way to mark NYI features. You should instead add a missing feature guard to `MissingFeatures.h`, and change code here to:

```cpp
if (!constOrdering) {
  assert(!cir::MissingFeatures::atomicVariableOrdering());
  return {};
}
```

Reply:
Lets make it a NYI error, I don't think this is important enough to create a Missingfeature flag